plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteTable.java

```diff
@@ -81,54 +82,54 @@ protected boolean isTruncateSupported() {
     // We use ROWID only if we don't have primary key. Looks like it is the only way to determine ROWID column presence.
     @Override
     public DBDPseudoAttribute[] getPseudoAttributes() throws DBException {
-        if (hasPrimaryKey()) {
+        if (hasPrimaryKey(this)) {
             return null;
         }
         return new DBDPseudoAttribute[] { PSEUDO_ATTR_ROWID };
     }
 
     @Override
     public DBDPseudoAttribute[] getAllPseudoAttributes(@NotNull DBRProgressMonitor monitor) throws DBException {
-        if (this.allPseudoAttributes == null) {
-
-            boolean isWithoutRowId = this.obtainIsWithoutRowId(monitor);
-            if (isWithoutRowId) {
-                this.allPseudoAttributes = DBDPseudoAttribute.EMPTY_ARRAY;
-            } else {
-                // see https://www.sqlite.org/lang_createtable.html#rowid (5. ROWIDs and the INTEGER PRIMARY KEY):
-                //     If a table contains a user defined column named "rowid", "oid" or "_rowid_",
-                //     then that name always refers the explicitly declared column and cannot be used to retrieve the integer rowid value.
-
-                Set<String> columnNames = this.getAttributes(monitor).stream()
-                    .map(a -> a.getName().toLowerCase())
-                    .collect(Collectors.toSet());
-                this.allPseudoAttributes = ALL_KNOWN_PSEUDO_ATTRS.stream()
-                    .filter(a -> !columnNames.contains(a.getName())) // all names are lowercased here
-                    .toArray(DBDPseudoAttribute[]::new);
-            }
+        return this.allPseudoAttributes != null ? this.allPseudoAttributes : (this.allPseudoAttributes = obtainAllPseudoAttributes(this.getDataSource(), this, monitor));
+    }
+
+    public static DBDPseudoAttribute[] obtainAllPseudoAttributes(GenericDataSource dataSource, DBSTable table, @NotNull DBRProgressMonitor monitor) throws DBException {
+        boolean isWithoutRowId = obtainIsWithoutRowId(dataSource, table, monitor);
+        if (isWithoutRowId) {
+            return DBDPseudoAttribute.EMPTY_ARRAY;
+        } else {
+            // see https://www.sqlite.org/lang_createtable.html#rowid (5. ROWIDs and the INTEGER PRIMARY KEY):
+            //     If a table contains a user defined column named "rowid", "oid" or "_rowid_",
+            //     then that name always refers the explicitly declared column and cannot be used to retrieve the integer rowid value.
+
+            Set<String> columnNames = table.getAttributes(monitor).stream()
```

Comment:
Two new sets + case conversion, seems to be over-complicated.
Can you just return ALL_KNOWN_PSEUDO_ATTRS filtered with column presence check? Like table.getAttribute(monitor, name)?